### PR TITLE
[azure_application_insights] [app_insights] Add dimensions and metric_type for metrics

### PR DESCRIPTION
**Pull request by ghost**
## What does this PR do?

Dimensions added:
- agent.id
- azure.dimensions.*
- azure.namespace
- cloud.region
- azure.resource.id
- azure.timegrain
- azure.dimensions.fingerprint

Added an ingest pipeline to create a fingerprint of the `azure.dimensions` field.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Relates #7616

### CODE REVIEW

File: packages/azure_application_insights/data_stream/app_insights/fields/package-fields.yml [hunk 1]
```diff
@@ -58,10 +61,16 @@
       description: >
         Azure metric dimensions.
 
+    - name: dimensions.fingerprint
+      type: keyword
+      dimension: true
+      description: |
+        Autogenerated ID representing the fingerprint of the azure.dimensions object
     - name: metrics.*.*
```
Comment (tetianakravchenko) [inline packages/azure_application_insights/data_stream/app_insights/fields/package-fields.yml:69]: where metrics are actually stored? in this field or in `azure.app_insights.metrics.*.*` ? I think one of them is redundant and should be removed

Comment (zmoog) [reply packages/azure_application_insights/data_stream/app_insights/fields/package-fields.yml:69]: According to some app insights metrics I collected in June, the app insight metricset stores metrics in the `azure.metrics.*` fields:

image

Comment (zmoog) [reply packages/azure_application_insights/data_stream/app_insights/fields/package-fields.yml:69]: @gpop63, did you get metrics in these fields as well in your recent tests?

**Review by zmoog — CHANGES_REQUESTED** [review]
We should double-check in which fields the metricset is storing metrics.

File: packages/azure_application_insights/data_stream/app_insights/fields/package-fields.yml (hunk 1 above)
Comment (gpop63) [reply packages/azure_application_insights/data_stream/app_insights/fields/package-fields.yml:69]: @zmoog Yeah, the metrics are stored in `azure.metrics` so the `metrics` from `fields.yml` could be removed.

Comment (kaiyan-sheng) [reply packages/azure_application_insights/data_stream/app_insights/fields/package-fields.yml:70]: Seems like metrics collected from `azure.monitor` also store fields under `azure.metrics`. Will that be a problem?

File: packages/azure_application_insights/changelog.yml [hunk 2]
```diff
@@ -1,3 +1,8 @@
+- version: "1.2.0"
```
Comment (tetianakravchenko) [inline packages/azure_application_insights/changelog.yml:1]: please change to `1.1.0` if merged before https://github.com/elastic/integrations/pull/7550

**Review by tetianakravchenko — APPROVED** [review]

**Review by zmoog — APPROVED** [review]

**Review by zmoog — APPROVED** [review]

File: packages/azure_application_insights/data_stream/app_insights/fields/package-fields.yml (hunk 1 above)
Comment (zmoog) [reply packages/azure_application_insights/data_stream/app_insights/fields/package-fields.yml:70]: Uhm, this should be okay. All metrics in the `azure.monitor` field seem fine.

**Pull request merged**